library/ssl_tls13_server.c

```diff
@@ -1276,6 +1276,37 @@ static int ssl_tls13_write_certificate_request( mbedtls_ssl_context *ssl )
 }
 #endif /* MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED */
 
+/*
+ * Handler for MBEDTLS_SSL_SERVER_CERTIFICATE
+ */
+static int ssl_tls13_write_server_certificate( mbedtls_ssl_context *ssl )
+{
+#if defined(MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED)
```

Comment:
Why not guarding with MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED the whole function ? Ditto for ssl_tls13_write_certificate_verify ? At the end of `ssl_tls13_write_encrypted_extensions` we have:
```C
#if defined(MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED)                             
    if( mbedtls_ssl_tls13_some_psk_enabled( ssl ) )                             
        mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_SERVER_FINISHED );    
    else                                                                        
        mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_CERTIFICATE_REQUEST );
#else                                                                           
    mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_SERVER_FINISHED );        
#endif
```